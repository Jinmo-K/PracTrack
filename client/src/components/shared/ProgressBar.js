import React from 'react';


const ProgressBar = ({ current, goal, showText }) => {
    const progress = current > goal ? 100 : Math.round(current / goal * 100);

    return (
        <div>
            <div className={current > 0 ? "progress border bg-light mb-2 mx-auto mx-sm-0" : "progress border bg-light mb-2 mx-auto"}>
                <div className="progress-bar bg-success progress-bar-striped-custom" role="progressbar" aria-valuenow={progress} aria-valuemin="0" aria-valuemax="100" style={{'width': progress + '%'}}>
                    {(showText) 
                        ? progress + '%' 
                        : null
                    }
                </div>
            </div>
        </div>
    );
};

export default ProgressBar;