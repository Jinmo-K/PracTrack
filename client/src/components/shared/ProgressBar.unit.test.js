import React from 'react';
import renderer from 'react-test-renderer';
import ProgressBar from './ProgressBar';

describe('<ProgressBar />', () => {
  it('should render correctly when goal has been achieved and text is shown', () => {
    const tree = renderer
      .create(<ProgressBar current={100} goal={10} showText={true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when goal has not been achieved and text is shown', () => {
    const tree = renderer
      .create(<ProgressBar current={1} goal={10} showText={true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when current progress is 0 and text is shown', () => {
    const tree = renderer
      .create(<ProgressBar current={0} goal={10} showText={true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when goal has been achieved and text is hidden', () => {
    const tree = renderer
      .create(<ProgressBar current={100} goal={10} showText={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when goal has not been achieved and text is hidden', () => {
    const tree = renderer
      .create(<ProgressBar current={1} goal={10} showText={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when current progress is 0 and text is hidden', () => {
    const tree = renderer
      .create(<ProgressBar current={0} goal={10} showText={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})
