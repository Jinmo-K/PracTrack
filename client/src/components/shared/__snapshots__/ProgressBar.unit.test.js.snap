// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar /> should render correctly when current progress is 0 and text is hidden 1`] = `
<div>
  <div
    className="progress border bg-light mb-2 mx-auto"
  >
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow={0}
      className="progress-bar bg-success progress-bar-striped-custom"
      role="progressbar"
      style={
        Object {
          "width": "0%",
        }
      }
    />
  </div>
</div>
`;

exports[`<ProgressBar /> should render correctly when current progress is 0 and text is shown 1`] = `
<div>
  <div
    className="progress border bg-light mb-2 mx-auto"
  >
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow={0}
      className="progress-bar bg-success progress-bar-striped-custom"
      role="progressbar"
      style={
        Object {
          "width": "0%",
        }
      }
    >
      0%
    </div>
  </div>
</div>
`;

exports[`<ProgressBar /> should render correctly when goal has been achieved and text is hidden 1`] = `
<div>
  <div
    className="progress border bg-light mb-2 mx-auto mx-sm-0"
  >
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow={100}
      className="progress-bar bg-success progress-bar-striped-custom"
      role="progressbar"
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;

exports[`<ProgressBar /> should render correctly when goal has been achieved and text is shown 1`] = `
<div>
  <div
    className="progress border bg-light mb-2 mx-auto mx-sm-0"
  >
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow={100}
      className="progress-bar bg-success progress-bar-striped-custom"
      role="progressbar"
      style={
        Object {
          "width": "100%",
        }
      }
    >
      100%
    </div>
  </div>
</div>
`;

exports[`<ProgressBar /> should render correctly when goal has not been achieved and text is hidden 1`] = `
<div>
  <div
    className="progress border bg-light mb-2 mx-auto mx-sm-0"
  >
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow={10}
      className="progress-bar bg-success progress-bar-striped-custom"
      role="progressbar"
      style={
        Object {
          "width": "10%",
        }
      }
    />
  </div>
</div>
`;

exports[`<ProgressBar /> should render correctly when goal has not been achieved and text is shown 1`] = `
<div>
  <div
    className="progress border bg-light mb-2 mx-auto mx-sm-0"
  >
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow={10}
      className="progress-bar bg-success progress-bar-striped-custom"
      role="progressbar"
      style={
        Object {
          "width": "10%",
        }
      }
    >
      10%
    </div>
  </div>
</div>
`;
